qutip/solver/heom/bofin_baths.py

```diff
@@ -973,10 +1057,621 @@ def _calc_eps(self, Nk):
+        # extend the list with the complex conjugates:
+        ckAI.extend(np.conjugate(ckAI))
+
+        vkAI = [-x - 1.0j * y for x, y in zip(gamma2, w02)]
+        vkAI.extend([-x + 1.0j * y for x, y in zip(gamma2, w02)])
+        self.ckAR, self.vkAR, self.ckAI, self.vkAI = [ckAR, vkAR, ckAI, vkAI]
+        self.Bath_corr = BosonicBath(self.Q, ckAR, vkAR, ckAI, vkAI)
+        self.exponents = self.Bath_corr.exponents
+
+
+class OhmicBath(BosonicBath):
+    def __init__(self, T, Q, alpha, wc, s, Nk=4, method="spectral", rmse=7e-5):
+        self.alpha = alpha
+        self.wc = wc
+        self.s = s
+        self.method = method
+        self.Q = Q
+        self.T = T
+        self.Nk = Nk
+        self.beta = 1 / T
+        if method == "correlation":
+            self.fit = FitCorr(self.Q)
+            t = np.linspace(0, 15 / self.wc, 1000)
+            C = self.ohmic_correlation(t, self.s)
+            self.fit.fit_correlation(t, C, final_rmse=rmse)
+            self.bath = self.fit.Bath_corr
+        else:
+            self.fit = FitSpectral(self.T, self.Q, self.Nk)
+            w = np.linspace(0, 15 * self.wc, 20000)
+            J = self.ohmic_spectral_density(w)
+            self.fit.get_fit(J, w, final_rmse=rmse)
+            self.bath = self.fit.Bath_spec
+            self.terminator = self.fit.terminator
+
+    def summary(self):
+        self.fit.summary()
+
+    def ohmic_spectral_density(self, w):
+        """The Ohmic bath spectral density as a function of w
+        (and the bath parameters).
+        """
+        return (
+            self.alpha
+            * w ** (self.s)
+            / (self.wc ** (1 - self.s))
+            * np.e ** (-abs(w) / self.wc)
+        )
+
+    def ohmic_correlation(self, t, s=1):
+        """The Ohmic bath correlation function as a function of t
+        (and the bath parameters).
+        """
+        corr = (
+            (1 / np.pi)
+            * self.alpha
+            * self.wc ** (1 - s)
+            * self.beta ** (-(s + 1))
+            * mp.gamma(s + 1)
+        )
+        z1_u = (1 + self.beta * self.wc - 1.0j *
+                self.wc * t) / (self.beta * self.wc)
+        z2_u = (1 + 1.0j * self.wc * t) / (self.beta * self.wc)
+        # Note: the arguments to zeta should be in as high precision
+        # as possible.
+        # See http://mpmath.org/doc/current/basics.html#providing-correct-input
+        return np.array(
+            [
+                complex(corr * (mp.zeta(s + 1, u1) + mp.zeta(s + 1, u2)))
+                for u1, u2 in zip(z1_u, z2_u)
+            ],
+            dtype=np.complex128,
+        )
+
+    def ohmic_power_spectrum(self, w):
+        """The Ohmic bath power spectrum as a function of w
+        (and the bath parameters).
+        """
+        return (
+            self.ohmic_spectral_density(w)
+            * ((1 / (np.e ** (w * self.beta) - 1)) + 1)
+            * 2
+        )
+
+
+# Fitting Functions
```

Comment:
I think you're right, specially if the AAA stuff ends up working, because then code just for fitting would get longer, not sure if I should move those now, or wait for a high level review on the structure of these things